src/libraries/System.DirectoryServices/src/System/DirectoryServices/ActiveDirectory/ADAMInstance.cs

```diff
@@ -518,7 +518,7 @@ public string HostName
                 if (_cachedHostName == null)
                 {
                     DirectoryEntry serverEntry = directoryEntryMgr.GetCachedDirectoryEntry(ServerObjectName);
-                    _cachedHostName = (string)PropertyManager.GetPropertyValue(context, serverEntry, PropertyManager.DnsHostName);
+                    _cachedHostName = (string)PropertyManager.GetPropertyValue(context, serverEntry, PropertyManager.DnsHostName)!;
```

Comment:
This property will always be non-null?

Reply:
I have no way of 100% knowing given this comes from the network but nothing is expecting this to be null, also many places in the code asking for `PropertyManager.DnsHostName` would throw in some way if they got null answer